app/models/calculator/flexi_rate.rb

```diff
@@ -25,7 +25,7 @@ def self.available?(_object)
 
     def compute(object)
       max = preferred_max_items.to_i
-      items_count = line_items_for(object).map(&:quantity).sum
+      items_count = line_items_for(object).map(&:quantity).to_a.sum
```

Comment:
If the objects are definitely line items, I think we can do `line_items_for(object).sum(:quantity)` in this case, and it will have much better performance.